antora site structure documentation update (#4655)
owncloud/docs: docs/antora-site-structure.md

## Scope of Documentation Repositories

The ownCloud documentation consists of a main repo named `docs` and additional product repos like the clients or others which are included as content sources defined in `site.yml`.

> Our setup is made in a way, where each repo can build it's own content individually for testing and checking validity. Only the build of main repo `docs` creates documentation which is pushed to the web. There is one exeption, pdf's are created AND pushed via the CI from _each_ content source.

Reviewer: ```suggestion
Our setup is made in a way, where each repo can build it's own content individually for testing and checking validity. Only the build of main repo `docs` creates documentation which is pushed to the web. There is one exception, PDF's are created AND pushed via the CI from _each_ content source.
```

### Scope of the playbook files (site.yml)

In general, only one `site.yml` file is neccessary for the whole environment and its definitions are **available to the whole site**. This `site.yml` file is located in the main repo `docs`. We have additionally for each content source its own `site.yml` for testing purposes only. The scope of these local `site.yml` files is restricted to the respective content source and any definitions made are not availabe outside. 

Due to this fact, you need to re-add all relevant attributes of the main `site.yml` file in the `site.yml` file of the content source which accesses it, else a local build will return warnings about unresolved attributes.

If you have added an attribute in a content source `site.yml` file, you must add this attibute to the main `site.yml` file to avoid a build warning (unresolved attribute) during a build of the entire documentation.

Note that this behaviour is relevant for the playbook `site.yml` files only and does not apply to the component descriptor files `antora.yml`.

## Scope of Content Accessibility

> Because Antora is capable of defining additional content sources, you can access content from these resources. The setup is flat, no main/child environment. To access ressources, follow the [Resource ID Coordinates][link-resource-id] scheme.

Reviewer: ```suggestion
Because Antora is capable of defining additional content sources, you can access content from these resources. The setup is flat, no main/child environment. To access resources, follow the [Resource ID Coordinates][link-resource-id] scheme.
```

> Because the setup we have made regarding testing, the direction of the arrow is important.

Reviewer: ```suggestion
Because of the setup we have made regarding testing, the direction of the arrow is important.
```

**Possible**
```
docs    --> docs-client-ios-app
            └> index.adoc
```

> Main (docs) can access content from any conten source (like docs-client-ios-app) and vice versa *at build time* , because docs has referenced the repo (content source) and made it available to all.

Reviewer: ```suggestion
Main (docs) can access content from any content source (like docs-client-ios-app) and vice versa *at build time* , because docs has referenced the repo (content source) and made it available to all.
```

**Impossible**
```
docs-client-ios-app    --> docs-client-desktop
                           └> index.adoc

docs-client-ios-app    --> docs
                           └> index.adoc

```

> When doing a build of a content source (like the docs-client-ios-app) which is neccesary for testing purposes, any references to another content source or to docs will fail as the referenced content source is unknown. Those references will throw an error/warning. Even not the best approach, use .html references to any other source outside the working repo.

Reviewer: ```suggestion
When doing a build of a content source (like the docs-client-ios-app) which is neccesary for testing purposes, any references to another content source or to docs will fail as the referenced content source is unknown. Those references will throw an error/warning. Even if it is not the best approach, use .html references to any other source outside the working repo.
```

While you can read more details about [What is antora.yml?][link-antora-yml] and [What is site.yml (the playbook)][link-site-yml], here are some important items:

> To manage versions in docs, we use branches. This means that any content based on a variable (attribute) limited to a branch must go into teh component descripto file `antora.yml` and be maintained accordingly. Any attribute that can be used in any branch of a component must be defined in `site.yml`

Reviewer: ```suggestion
To manage versions in docs, we use branches. This means that any content based on a variable (attribute) limited to a branch must go into the component description file `antora.yml` and be maintained accordingly. Any attribute that can be used in any branch of a component must be defined in `site.yml`
```
